packages/field-base/src/field-mixin.js

```diff
@@ -97,55 +93,89 @@ export const FieldMixin = (superclass) =>
 
       // Save generated ID to restore later
       this.__savedHelperId = this._helperId;
+
+      this.__helperIdObserver = new MutationObserver((mutationRecord) => {
+        mutationRecord.forEach((mutation) => {
+          // only handle helper nodes
+          if (
+            mutation.type === 'attributes' &&
+            mutation.attributeName === 'id' &&
+            mutation.target === this._currentHelper &&
+            mutation.target.id !== this.__savedHelperId
+          ) {
+            this.__updateHelperId(mutation.target);
+          }
+        });
+      });
     }
 
     /** @protected */
     ready() {
       super.ready();
 
-      if (this._errorNode) {
-        this._errorNode.id = this._errorId;
+      const error = this._errorNode;
+      if (error) {
+        error.id = this._errorId;
 
-        this._applyCustomError();
+        this.__applyCustomError();
 
         this._updateErrorMessage(this.invalid, this.errorMessage);
       }
 
-      if (this._helperNode) {
-        this._currentHelper = this._helperNode;
-        this._helperNode.id = this._helperId;
-
-        this._applyCustomHelper();
-
-        this.__helperSlot = this.shadowRoot.querySelector('[name="helper"]');
-        this.__helperSlot.addEventListener('slotchange', this.__onHelperSlotChange.bind(this));
-        this.__helperIdObserver = new MutationObserver((mutationRecord) => {
-          mutationRecord.forEach((mutation) => {
-            // only handle helper nodes
-            if (
-              mutation.type === 'attributes' &&
-              mutation.attributeName === 'id' &&
-              mutation.target === this._currentHelper &&
-              mutation.target.id !== this.__savedHelperId
-            ) {
-              this.__updateHelperId(mutation.target);
-            }
-          });
-        });
-
-        this.__helperIdObserver.observe(this, { attributes: true, subtree: true });
+      const helper = this._helperNode;
+      if (helper) {
+        this.__applyCustomHelper(helper);
       }
+
+      this.__helperSlot = this.shadowRoot.querySelector('[name="helper"]');
+
+      this._observer = new FlattenedNodesObserver(this.__helperSlot, (info) => {
+        const helper = this._currentHelper;
+
+        const newHelper = info.addedNodes.find((node) => node !== helper);
+        const oldHelper = info.removedNodes.find((node) => node === helper);
+
+        if (newHelper) {
+          // Custom helper is added, remove the default one.
+          if (helper && helper.isConnected) {
+            this.removeChild(helper);
+          }
+
+          this.__applyCustomHelper(newHelper);
+
+          this.__helperIdObserver.observe(newHelper, { attributes: true, subtree: true });
+        } else if (oldHelper && this.helperText) {
+          // Custom helper is removed, restore the default one.
+          const helper = this.__defaultHelper;
+          helper.textContent = this.helperText;
```

Comment:
We probably should ensure that the old node is no longer observed in case the node has been rather moved to a different place than completely removed from the DOM.

Reply:
Thanks, I will fix this on Monday.

Reply:
Done. Currently there is no support for `unobserve` in MutationObserver: https://github.com/whatwg/dom/issues/126
So I had to change code to create new observer every time, see 7fa40ab